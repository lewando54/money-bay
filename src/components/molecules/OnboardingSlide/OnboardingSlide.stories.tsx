/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { StoryObj, Meta } from '@storybook/react-native'

import OnboardingSlide from './OnboardingSlide'

const meta = {
    component: OnboardingSlide,
    argTypes: {
        // To write
    }
} satisfies Meta<typeof OnboardingSlide>

export default meta
type Story = StoryObj<typeof OnboardingSlide>

export const Default: Story = {
    args: {
        // To write
    }
}
