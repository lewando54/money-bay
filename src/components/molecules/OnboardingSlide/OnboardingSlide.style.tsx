import { Dimensions, StyleSheet } from 'react-native'

import globalThemeStyle, { BODY_TEXT_COLOR, MAIN_DARK } from '@styling/GlobalTheme.style'

const {width, height} = Dimensions.get('screen')

export default StyleSheet.create({
    mainContainer: {
        alignItems: 'center',
        justifyContent: 'flex-start',
        width,
        height: height * 0.77,
        marginTop: height * 0.08
    },
    image: {
        width: width * 0.90,
        height: height * 0.39,
        resizeMode: 'contain',
    },
    textContainer: {
        width: width * 0.8933,
        alignItems: 'center',
        gap: 20,
        paddingHorizontal: 20,
        paddingTop: height * 0.06,
        overflow: 'hidden'
    },
    heading: {
        ...globalThemeStyle.text_Bold,
        fontSize: 28,
        color: MAIN_DARK,
        textAlign: 'center'
    },
    body: {
        ...globalThemeStyle.text_Regular,
        fontSize: 16,
        color: BODY_TEXT_COLOR,
        textAlign: 'center',
        paddingHorizontal: 6
    }
})
