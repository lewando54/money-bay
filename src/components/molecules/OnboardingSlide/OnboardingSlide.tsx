import React from 'react'
import onboardingSlideStyle from './OnboardingSlide.style'
import { Dimensions, Image, ImageSourcePropType, StyleProp, Text, View, ViewStyle } from 'react-native'
import Logo from './assets/Logo.svg'
import Animated, { Extrapolate, SharedValue, interpolate, useAnimatedStyle } from 'react-native-reanimated'

interface IOnboardingSlideProps {
    id?: number
    index?: number
    imageSource?: ImageSourcePropType
    style?: StyleProp<ViewStyle>
    heading?: string
    opacity?: SharedValue<number>
    body?: string
    testID?: string
}

export default function OnboardingSlide({
    id,
    index,
    imageSource=require('./assets/onboarding-2.png'),
    style,
    heading='Welcome to MoneyBay app!',
    opacity,
    body='Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.',
    testID
}: IOnboardingSlideProps){
    const {width} = Dimensions.get('window')

    const opacityAnimation = opacity && useAnimatedStyle(() => {
        const opacityV = interpolate(
            opacity.value, 
            [
                (index - 1) * width,
                index * width,
                (index + 1) * width
            ],
            [0, 1, 0],
            Extrapolate.CLAMP
        )

        return {
            opacity: opacityV
        }
    })

    return (
        <Animated.View testID={testID} style={[onboardingSlideStyle.mainContainer, style, opacity && opacityAnimation]} key={id}>
            <Image source={imageSource} style={onboardingSlideStyle.image}/>
            <View style={onboardingSlideStyle.textContainer}>
                <Logo />
                <Text style={onboardingSlideStyle.heading}>{heading}</Text>
                <Text style={onboardingSlideStyle.body}>{body}</Text>
            </View>
        </Animated.View>
    )
}
