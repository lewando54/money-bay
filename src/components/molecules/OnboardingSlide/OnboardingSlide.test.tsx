import React from 'react'
import renderer from 'react-test-renderer'
import OnboardingSlide from './OnboardingSlide'
describe('OnboardingSlide', () => {
    it('should render correctly', () => {
        const tree = renderer.create(<OnboardingSlide />)
        expect(tree).toMatchSnapshot()
    })
})
