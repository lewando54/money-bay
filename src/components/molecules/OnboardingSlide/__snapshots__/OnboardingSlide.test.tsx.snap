// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OnboardingSlide should render correctly 1`] = `
<View
  animatedStyle={
    {
      "value": {},
    }
  }
  collapsable={false}
  style={
    {
      "alignItems": "center",
      "height": 1027.18,
      "justifyContent": "flex-start",
      "marginTop": 106.72,
      "width": 750,
    }
  }
>
  <Image
    source={1}
    style={
      {
        "height": 520.26,
        "resizeMode": "contain",
        "width": 675,
      }
    }
  />
  <View
    style={
      {
        "alignItems": "center",
        "gap": 20,
        "overflow": "hidden",
        "paddingHorizontal": 20,
        "paddingTop": 80.03999999999999,
        "width": 669.975,
      }
    }
  >
    <SvgMock />
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishBold",
            "fontSize": 28,
            "textAlign": "center",
          },
        ]
      }
    >
      Welcome to MoneyBay app!
    </Text>
    <Text
      style={
        [
          {
            "color": "#1B1D4D",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
          },
          {
            "color": "#4C4C60",
            "fontFamily": "MulishRegular",
            "fontSize": 16,
            "paddingHorizontal": 6,
            "textAlign": "center",
          },
        ]
      }
    >
      Labore sunt culpa excepteur culpa ipsum. Labore occaecat ex nisi mollit.
    </Text>
  </View>
</View>
`;
